bin/build/src/metabuild_common/output.clj

```diff
@@ -4,7 +4,12 @@
    [clojure.string :as str]
    [colorize.core :as colorize]))
 
-(def ^:dynamic *steps* [])
+(set! *warn-on-reflection* true)
+
+(def ^:dynamic *steps*
+  "Vector of all the parent steps/substeps we're currently in the process of working on. (See [[metabuild-common.steps]]
+  for more info."
+  [])
```

Comment:
Thanks to Cam Saul for [this nice docstring](https://github.com/metabase/metabase/commit/90b876c847d782a9b64d662de0e676c41154e0a8#diff-c7d2bc4f60e7aef7c28bbc8c4f4a598bf2cff23c39e7355acbe750643c7d786bR9-R12), and about half of this PR.